### PR TITLE
Fix/nodeid bug

### PR DESCRIPTION
**Pull request by Keyrxng**
Resolves https://github.com/ubiquity/devpool-directory-bounties/issues/42

- adds logic to remove mismatch label issues before beginning any state changing effects
- adds logic to automate the `DEVPOOL_OWNER_NAME` variable as it's annoying af having to change it back and forth all of the time
- `deleteUnauthorizedIssues` now fetches `all` issues and also deletes all duplicate issues based on the devpool task body.

Both our devpools share the same number of issues at the moment but I can't validate the body url changing without running it from this repo.

https://github.com/ubq-testing/devpool-directory/actions/runs/10977087962/job/30478644205

### CODE REVIEW

File: .env.example [hunk 1]
```diff
@@ -1,6 +1,18 @@
...
+
+// this is set by the workflow, for local testing set it to `ubiquity/devpool-directory`
+OWNER_REPO=yourOrg/yourForkName
+// these are used to delete unauthorized issues
+APP_PRIVATE_KEY=
+APP_ID=
+// this one needs to belong to an admin of the devpool-directory repo in order to delete issues.
+GH_TOKEN=
+
+// the jest test environment
```
Comment (0x4007) [inline .env.example:17]:
```suggestion
# the jest test environment
```
Wrong comment syntax

File: handlers/state-updates.ts [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+    }
+
+    console.log(`Updated metadata: ${devpoolIssue.html_url} - (${projectIssue.html_url})`, metaChanges);
+  }
+}
+
+export async function applyStateChanges(missingInPartners: boolean, projectIssue: GitHubIssue, devpoolIssue: GitHubIssue, taskHasPriceLabel: boolean) {
+  const stateChanges: StateChanges = {
+    // missing in the partners
+    forceMissing_Close: {
```
Comment (0x4007) [inline handlers/state-updates.ts:42]: Strange naming convention is this from GitHub? Otherwise you should fix

File: handlers/state-updates.ts [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+      cause: missingInPartners,
+      effect: "closed",
+      comment: "Closed (missing in partners)",
```
Comment (0x4007) [inline handlers/state-updates.ts:45]: Shouldn't rely on posting comments although I'm not sure what this is.

**Review by 0x4007 — CHANGES_REQUESTED** [review]
Post QA?

File: handlers/state-updates.ts (hunk 3 above)
Comment (Keyrxng) [reply handlers/state-updates.ts:45]: comment here is for the logs only

File: handlers/state-updates.ts (hunk 2 above)
Comment (Keyrxng) [reply handlers/state-updates.ts:42]: It's to clearly signify the action which is being performed at a glance. It has been this way for some time, but i assume just drop the "_" is acceptable?

**Comment by Keyrxng** [comment]
> Post QA?

My runs are filled other QA examples but this is the most recent one which has the same final task output as the devpool

https://github.com/ubq-testing/devpool-directory/actions/runs/10977087962/job/30478644205

**Pull request closed without merging**